fv3gfs/physics/testing/translate_physics.py

```diff
@@ -0,0 +1,69 @@
+from fv3core.testing import TranslateFortranData2Py
+import numpy as np
+import fv3core.utils.gt4py_utils as utils
+
+
+def read_serialized_data(serializer, savepoint, variable):
+    data = serializer.read(variable, savepoint)
+    n_dim = len(data.shape)
+    cn = int(np.sqrt(data.shape[0]))
+    npz = data.shape[1]
+    if n_dim == 2:
+        var_reshape = np.reshape(data, (cn, cn, npz))
+        rearranged = var_reshape[:, :, ::-1]
+    elif n_dim == 3:
+        n_trac = data.shape[2]
+        var_reshape = np.reshape(data, (cn, cn, npz, n_trac))
+        rearranged = var_reshape[:, :, ::-1, :]
+    elif len(data.flatten()) == 1:
+        rearranged = data[0]
+    else:
+        raise NotImplementedError("Data dimension not supported")
+    return rearranged
+
+
+class TranslatePhysicsFortranData2Py(TranslateFortranData2Py):
+    def __init__(self, grid):
+        super().__init__(grid)
+
+    def collect_input_data(self, serializer, savepoint):
+        input_data = {}
+        for varname in (
+            self.serialnames(self.in_vars["data_vars"]) + self.in_vars["parameters"]
+        ):
+            input_data[varname] = read_serialized_data(serializer, savepoint, varname)
+        return input_data
+
+    def slice_output(self, inputs, out_data=None):
+        utils.device_sync()
```

Comment:
this may not longer be necessary, but I guess doesn't hurt.

Reply:
Removed device sync since it's not required